ext/opcache/jit/zend_jit_x86.dasc

```diff
@@ -9388,11 +9388,59 @@ static int zend_jit_init_fcall(dasm_State **Dst, const zend_op *opline, uint32_t
 			|	mov r0, EX->func
 		}
 	} else {
-		|	// if (CACHED_PTR(opline->result.num))
+		|	// CACHED_PTR(opline->result.num)
 		|	mov r2, EX->run_time_cache
 		|	mov r0, aword [r2 + opline->result.num]
-		|	test r0, r0
-		|	jz >1
```

Comment:
I checked for run_time_cache entry on purpose. If it's already initialized, we may omit the guard check, because non-polimorphic functions can't be changed during request.

Reply:
During the second request, when the function has changed, we exit the trace at [[0]](https://github.com/php/php-src/blob/106c908/ext/opcache/jit/zend_jit_x86.dasc#L9360). We then execute the trace again during the second iteration of the loop. This time, the run_time_cache is initialized (it has been initialized by the VM after the exit), so we continue directly to >3.

The test `gh8461-005.phpt` demonstrates this.

Reply:
Thanks for explanation. I plan to work on this later today or tomorrow.

Reply:
@arnaud-lb please take a look at https://github.com/php/php-src/pull/8535. do you see any problems?

Reply:
Looks good to me, thanks :)